File: leetcode/javascript/1. Two Sum.js
var twoSum = function(nums, target) {
    const answer = [];
    const diff = {};

    for (let i = 0; i < nums.length; i += 1) {
        if (!diff[nums[i]]) {
            diff[nums[i]] = {
                index: i,
                diffValue: target - nums[i],
            }
        } 
    }

    for (let i = 0; i < nums.length; i += 1) {
        const checkDiffValue = diff[nums[i]].diffValue;

        if (diff[checkDiffValue] && diff[checkDiffValue].index !== i) {
            answer.push(i, diff[checkDiffValue].index);
            break;
        }
    }

    return answer;
};

File: leetcode/python/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        answer = []
        diff = dict()

        for i in range(len(nums)):
            if nums[i] not in diff:
                diff[nums[i]] = [i, target - nums[i]]
            

        for i in range(len(nums)):
            checkDiffValue = target - nums[i]

            if checkDiffValue in diff and diff[checkDiffValue][0] != i:
                answer.append(i)
                answer.append(diff[checkDiffValue][0])
                break

        return answer